src/components/FallenHeroes.tsx

```diff
@@ -11,32 +11,46 @@ function FallenHeroes() {
   // matching hero from the set of totalHeroes,
   // and is included in rest of heroes alive, update the state accordingly
   const { campaign, setCampaign } = React.useContext(LotrContext);
+
+  function filterFallen() {
+    return campaign.allHeroes.filter((hero) => hero.alive === false);
+  }
+
+  const fallen = filterFallen();
   const [fallenHero, setFallenHero] = useState("");
 
+  const names = campaign.allHeroes.map((hero) => {
+    return hero.name;
+  });
+
+  const isAlive =
```

Comment:
`isAlive` and `names` are only used inside `addFallenHero`, no need to have them in the outer scope. It will be cleaner to move them inside.